iocage_lib/ioc_fetch.py

```diff
@@ -710,40 +711,52 @@ def fetch_download(self, _list, missing=False):
-
-                            if progress != last_progress:
-                                text = self.update_progress(
-                                    progress,
-                                    f'Downloading: {f}',
-                                    elapsed,
-                                    chunk_size
-                                )
-
-                                if progress % 10 == 0:
-                                    # Not for user output, but for callback
-                                    # heartbeats
-                                    iocage_lib.ioc_common.logit(
-                                        {
-                                            'level': 'INFO',
-                                            'message': text.rstrip()
-                                        },
-                                        _callback=self.callback,
-                                        silent=True)
-
-                            last_progress = progress
-                            start = time.time()
+                    try:
+                        for i, chunk in enumerate(
+                            r.iter_content(chunk_size=chunk_size), 1
+                        ):
+                            if chunk:
+                                elapsed = time.time() - start
+                                dl_progress += len(chunk)
+                                txz.write(chunk)
+
+                                progress = float(i) / float(total)
+                                if progress >= 1.:
+                                    progress = 1
+                                progress = round(progress * 100, 0)
+
+                                if progress != last_progress:
+                                    text = self.update_progress(
+                                        progress,
+                                        f'Downloading: {f}',
+                                        elapsed,
+                                        chunk_size
+                                    )
+
+                                    if progress % 10 == 0:
+                                        # Not for user output, but for callback
+                                        # heartbeats
+                                        iocage_lib.ioc_common.logit(
+                                            {
+                                                'level': 'INFO',
+                                                'message': text.rstrip()
+                                            },
+                                            _callback=self.callback,
+                                            silent=True)
+
+                                last_progress = progress
+                                start = time.time()
+                    except requests.exceptions.ChunkedEncodingError as exc:
+                        if isinstance(exc.__cause__, urllib3.exceptions.ProtocolError):
```

Comment:
Do we need this ? If we are seeing any kind of error related to connection, let's just raise an exception appropriately using iocage

Reply:
`ChunkedEncodingError` could be caused by malformed response, are we good saying that an internet connection issue too OR have separate feedback based on underlying `__cause__`?